### PR TITLE
8279936: Change shared code to use os:: system API's

### PR DESCRIPTION
**Pull request by hseigel**
Please review this small change to call os:: API's.  the changes were tested with Mach5 tiers 1-2 on Linux, Mac OS, and Windows, and Mach5 tiers 3-5 on Linux x64.  Most of these changes were for I/O related calls.  Changes to memory allocation calls such as malloc and free will be handled in a future change.

Thanks, Harold

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8279936](https://bugs.openjdk.java.net/browse/JDK-8279936): Change shared code to use os:: system API's

### Reviewers
 * [David Holmes](https://openjdk.java.net/census#dholmes) (@dholmes-ora - **Reviewer**)
 * [Kim Barrett](https://openjdk.java.net/census#kbarrett) (@kimbarrett - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/7055/head:pull/7055` \
`$ git checkout pull/7055`

Update a local copy of the PR: \
`$ git checkout pull/7055` \
`$ git pull https://git.openjdk.java.net/jdk pull/7055/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 7055`

View PR using the GUI difftool: \
`$ git pr show -t 7055`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/7055.diff">https://git.openjdk.java.net/jdk/pull/7055.diff</a>

</details>

### CODE REVIEW

File: test/hotspot/gtest/gtestMain.cpp [hunk 1]
```diff
@@ -278,7 +278,7 @@ static void runUnitTestsInner(int argc, char** argv) {
     int ret;
     if ((ret = init_jvm(argc, argv, is_vmassert_test, &jvm)) != 0) {
       fprintf(stderr, "ERROR: JNI_CreateJavaVM failed: %d\n", ret);
-      abort();
+      os::abort();
```
Comment (dholmes-ora) [inline test/hotspot/gtest/gtestMain.cpp:281]: No this should be the native abort function. We failed to create the VM so we don't/can't use any VM service.

**Review by dholmes-ora — APPROVED** [review]
Hi Harold,

This seems fine except for one change - see below.

I still think we need to discuss whether os::close and os::read should even exist but that is a separate matter.

Thanks,
David

**Review by dholmes-ora — APPROVED** [review]
Looks good!

Thanks,
David

**Comment by hseigel** [comment]
Thanks David and Kim for the reviews!

/integrate

**Pull request closed without merging**